content/5_Plat/Sweep_Line.mdx

```diff
@@ -162,8 +162,160 @@ For each point, we recalculate $\delta = \min(\delta, \delta_p)$, and update our
 <FocusProblem problem="segSam" />
 
 ### Solution
+Let's simplify the problem a little bit and focus on finding any overlapping segments. The segment we are looking for is one of the two.
 
-<IncompleteSection />
+To find a pair of overlapping segments we use a sweep line approach. Imagine sweeping a vertical line across the scene from left to right, pausing at every segment endpoint.
+We simulate this by sorting all the segment endpoints by x and walking through this sorted array (called 'events' in the code below). As we scan, we keep track of active segments
+by using a set (called 'active_segments' in the code below). When we hit the beginning point of a segment, we add it to the active set, and when we hit the ending point of a segment, we remove it from the active set.
+The operation of inserting or removing takes $\mathcal{O}(\log n)$.
+
+The active set of segments is ordered by $y$ coordinate. If two segments overlap, it means that they are adjacent in the set. So every time we insert or remove
```

Comment:
```suggestion
The active set of segments is ordered by $y$ coordinate. If two segments overlap, they are adjacent in the set., so every time we insert or remove
```

Reply:
Move this to the previous paragraph or smth: The active set of segments is ordered by $y$ coordinate.

Reply:
Ok